Slots App/app/src/main/java/com/example/challengeslotsapp/MainActivity.kt

```diff
@@ -1,24 +1,332 @@
+            }
+        }
+    }
+
+    @ExperimentalFoundationApi
+    @Composable
+    private fun PortraitOrientation(
+        gameMode: GameMode,
+        slotItems: List<Slot>
+    ) {
+        val rows = gameMode.rows
+        val columns = gameMode.columns
+        var slots by remember {
+            mutableStateOf(List(gameMode.rows) { slotItems })
+        }
+        var credits by remember {
+            mutableStateOf(1000)
+        }
+        val maxWin = 70
+        val minWin = 10
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = 15.dp, vertical = 50.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.SpaceAround
+        ) {
+            Title()
+            Credits(credits)
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+            ) {
+                Slots(slots, rows, columns)
+            }
+            Spin(
+                spin = {
+                    val spunSlots = SlotsGenerator(slots, slotItems).generateRandomSlots()
+
+                    val (wonSlotsCoordinates, amountOfWonLines) =
+                        if (gameMode.name == "Line") SlotsWinningCombinationsChecker(spunSlots).checkLine()
+                        else SlotsWinningCombinationsChecker(spunSlots).checkSquare()
+
+                    slots = WinningPositionsSetter(
+                        wonSlotsCoordinates,
+                        spunSlots
+                    ).setWinningPositions()
+
+                    credits -= 100
+                    credits += amountOfWonLines * (Random.nextInt(maxWin - minWin) + minWin)
+                }
+            )
+        }
+    }
+
+    @ExperimentalFoundationApi
+    @Composable
+    private fun LandscapeOrientation(
+        gameMode: GameMode,
+        slotItems: List<Slot>
+    ) {
+        val rows = gameMode.rows
+        val columns = gameMode.columns
+        var slots by remember {
+            mutableStateOf(List(gameMode.rows) { slotItems })
+        }
+        var credits by remember {
+            mutableStateOf(1000)
+        }
+        val maxWin = 70
+        val minWin = 10
+
+        Row(
+            modifier = Modifier.fillMaxSize(),
+            horizontalArrangement = Arrangement.SpaceAround,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Box(
+                modifier = Modifier
+                    .fillMaxHeight()
+                    .fillMaxWidth(0.5f),
+                contentAlignment = Alignment.Center
+            ) {
+                Slots(slots, rows, columns)
+            }
+
+            Column(
+                modifier = Modifier
+                    .fillMaxHeight(),
+                verticalArrangement = Arrangement.SpaceAround
+            ) {
+                Title(isLandscape = true)
+                Credits(credits = credits)
+                Spin(
```

Comment:
Yes, you are right, we can

Reply:
We can, but I'm not sure that it is the best solution in this case, because of a large number of parameters we have to pass. In my opinion, it will be better to create a separate class for `slots` and implement `spin` function as a method. How do you think?